File: frontend/src/app/service/BaseMetrics.ts
export class BaseMetrics {
  axioms: number| undefined;
  logicalAxiomsCount: number| undefined;
  classCount: number| undefined;
  anonymousClassCount: number| undefined;
  objectPropertyCount: number| undefined;
  dataPropertyCount: number| undefined;
  individualsCount: number| undefined;
  dlExpresivity: string| undefined;
  directImports: Set<string>| undefined;
  declaredImports: Set<string>| undefined;
  actualImports: Set<string>| undefined;
}

File: onto_generator/src/main/java/com/example/onto_generator/model/BaseMetrics.java
package com.example.onto_generator.model;


import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.RequiredArgsConstructor;
import org.semanticweb.owlapi.model.OWLOntology;

import java.util.Set;

@Builder
@AllArgsConstructor
@RequiredArgsConstructor
@Data
public class BaseMetrics {
    private int Axioms;
    private int LogicalAxiomsCount;
    private int ClassCount;
    private int AnonymousClassCount;
    private int ObjectPropertyCount;
    private int DataPropertyCount;
    private int IndividualsCount;
    private String DLExpresivity;
    private Set<String> directImports;
    private Set<String> declaredImports;
    private Set<String> actualImports;
}
